pkg/apply/apply.go

```diff
@@ -48,18 +49,17 @@ func ApplyObject(ctx context.Context, client k8sclient.Client, obj *uns.Unstruct
 		return errors.Wrapf(err, "could not retrieve existing %s", objDesc)
 	}
 
-	// Merge the desired object with what actually exists
-	if err := MergeObjectForUpdate(existing, obj); err != nil {
-		return errors.Wrapf(err, "could not merge object %s with existing", objDesc)
-	}
+	if !equality.Semantic.DeepEqual(existing, obj) {
+		// Merge the desired object with what actually exists
+		if err := MergeObjectForUpdate(existing, obj); err != nil {
```

Comment:
I'm pretty sure you have to call `MergeObjectForUpdate` before calling `DeepEqual` or it will always end up being unequal.

/hold
until this can get tested